builder/src/chunkdict_generator.rs

```
@@ -90,6 +188,211 @@ impl Generator {
+        blobtable: &mut RafsV6BlobTable,
+    ) -> Result<()> {
+        let mut bootstrap_ctx = bootstrap_mgr.create_ctx()?;
+        let mut bootstrap = Bootstrap::new(tree.clone())?;
+
+        // Build bootstrap
+        bootstrap.build(ctx, &mut bootstrap_ctx)?;
+
+        // Verify and update prefetch blob
+        assert!(
+            blobtable
+                .entries
+                .iter()
+                .filter(|blob| blob.blob_id() == "Prefetch-blob")
+                .count()
+                == 1,
+            "Expected exactly one Prefetch-blob"
+        );
+
+        // Rewrite prefetch blob id
+        blobtable
+            .entries
+            .iter_mut()
+            .filter(|blob| blob.blob_id() == "Prefetch-blob")
+            .for_each(|blob| {
+                let mut info = (**blob).clone();
+                info.set_blob_id(ctx.blob_id.clone());
+                *blob = Arc::new(info);
+            });
+
+        // Dump bootstrap
+        let blob_table_withprefetch = RafsBlobTable::V6(blobtable.clone());
+        bootstrap.dump(
+            ctx,
+            &mut bootstrap_mgr.bootstrap_storage,
+            &mut bootstrap_ctx,
+            &blob_table_withprefetch,
+        )?;
+
+        Ok(())
+    }
+
+    fn finalize_blob(
+        ctx: &mut BuildContext,
+        blobtable: &mut RafsV6BlobTable,
+        blob_state: &mut PrefetchBlobState,
+    ) {
+        blobtable.entries.push(blob_state.blob_info.clone().into());
+        let mut blob_mgr = BlobManager::new(nydus_utils::digest::Algorithm::Blake3);
+        blob_mgr.add_blob(blob_state.blob_ctx.clone());
+        blob_mgr.set_current_blob_index(0);
+        Blob::finalize_blob_data(&ctx, &mut blob_mgr, blob_state.blob_writer.as_mut()).unwrap();
+        if let Some((_, blob_ctx)) = blob_mgr.get_current_blob() {
+            Blob::dump_meta_data(&ctx, blob_ctx, blob_state.blob_writer.as_mut()).unwrap();
+        } else {
+            panic!();
```

Should return an specified error.